app/src/main/java/com/csai/swanton/tasks/ConnectTask.java

```diff
@@ -0,0 +1,53 @@
+package com.csai.swanton.tasks;
+
+import android.os.AsyncTask;
+import android.util.Log;
+import android.view.View;
+import com.csai.swanton.sftp.ChannelSftpFactory;
+import com.csai.swanton.util.ActivityTags.Tags;
+import com.google.android.material.snackbar.Snackbar;
+import com.jcraft.jsch.ChannelSftp;
+import com.jcraft.jsch.JSchException;
+import java.lang.ref.WeakReference;
+import java.util.Optional;
+import javax.annotation.Nonnull;
+import lombok.AllArgsConstructor;
+
+/**
+ * Asynchronous task for connecting to a raspberry pi.
+ */
+@AllArgsConstructor
+public class ConnectTask extends AsyncTask<Void, Void, Optional<ChannelSftp>> {
+
+  @Nonnull
+  private final WeakReference<View> view;
+
+  @Override
+  protected void onPreExecute() {
+    Snackbar.make(this.view.get(), "Connecting to Raspberry Pi...", Snackbar.LENGTH_SHORT).show();
+    Log.i(Tags.MAIN_ACTIVITY.getTag(), "Connecting to Raspberry Pi...");
```

Comment:
Maybe we should include the IP we're trying to connect to in the log

Reply:
Good point!